From cd2110bff81a22b38302d43fcc4c917ee0af2589 Mon Sep 17 00:00:00 2001
From: Lei Chen <ls.cat@outlook.com>
Date: Mon, 5 Dec 2022 20:06:30 +0800
Subject: [PATCH 4/4] chore: update Makefile

---
 Makefile.am | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 4689caa..45c797e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -81,18 +81,20 @@ memcached_CPPFLAGS = -DNDEBUG
 memcached_debug_LDADD = @PROFILER_LDFLAGS@
 memcached_debug_CFLAGS = @PROFILER_FLAGS@
 
-paris_PATH = /ssd/chenlei/atlas/Paris
-paris_INC_FLAG = -I${paris_PATH}/runtime/include -I${paris_PATH}/bks_module/include 
-paris_LIB_FLAG = -L${paris_PATH}/build/runtime/lib
+ATLAS_PATH = /ssd/chenlei/atlas/Paris
+JEMALLOC_DIR = $(ATLAS_PATH)/third_party/jemalloc
+ATLAS_INC = -I$(ATLAS_PATH)/runtime/include -I$(ATLAS_PATH)/bks_module/include -I$(ATLAS_PATH)/third_party/jemalloc/include
+ATLAS_LIB = -L$(ATLAS_PATH)/build/runtime/lib -L$(JEMALLOC_DIR)/lib -Wl,-rpath=$(ATLAS_PATH)/build/runtime/lib -Wl,-rpath=$(JEMALLOC_DIR)/lib
+
 memcached_LDADD = 
 memcached_DEPENDENCIES =
 memcached_debug_DEPENDENCIES =
 CLEANFILES=
 
-memcached_CFLAGS = ${paris_INC_FLAG} ${paris_LIB_FLAG}
-memcached_LDADD += -lparis
-memcached_debug_CFLAGS += ${paris_INC_FLAG} ${paris_LIB_FLAG}
-memcached_debug_LDADD += -lparis
+memcached_CFLAGS = ${ATLAS_INC} ${ATLAS_LIB}
+memcached_LDADD += -lparis -ljemalloc
+memcached_debug_CFLAGS += ${ATLAS_INC} ${ATLAS_LIB}
+memcached_debug_LDADD += -lparis -ljemalloc
 
 if BUILD_LINUX_PRIVS
 memcached_LDADD += -lseccomp
-- 
2.25.1

